skills/declarative/calendarSkill/runtime/customaction/MSGraphClient.cs

```diff
@@ -27,7 +33,9 @@ public static GraphServiceClient GetAuthenticatedClient(string accessToken, Http
 
         public static Exception HandleGraphAPIException(ServiceException ex)
         {
-            return new Exception($"Microsoft Graph API Exception: {ex.Message}");
+            // TODO: Why do we throw a very generic exception when there is a much better
```

Comment:
I believe this is a relic of the original C# implementation. We can take it out/modify this logic however we see fit. I think the original use of handling the ServieException differently was to be able to throw a specific message to the user like "I'm having trouble accessing Microsoft Graph" or something rather than "Sorry, something went wrong" but we haven't implemented that in this design